Core/Core/AppEnvironment/ExperimentalFeature.swift

```diff
@@ -25,48 +25,54 @@ import Foundation
-            return hosts.contains(host)
+        get { return enabled }
+        set {
+            enabled = newValue
+            UserDefaults.standard.set(newValue, forKey: self.settingsKey)
         }
     }
 
-    public static var allEnabled: Bool {
-        get { return UserDefaults.standard.bool(forKey: "ExperimentalFeature.allEnabled") }
-        set { UserDefaults.standard.set(newValue, forKey: "ExperimentalFeature.allEnabled") }
+    public static func settingsKey(forConfigKey key: String) -> String {
+        return "ExperimentalFeature.\(key)"
     }
 }
 
 extension ExperimentalFeature {
-    public static let assignmentListGraphQL = ExperimentalFeature(state: .disabled)
-    public static let parent3 = ExperimentalFeature(state: .disabled)
-    public static let conferences = ExperimentalFeature(state: .disabled)
-    public static let favoriteGroups = ExperimentalFeature(state: .disabled)
-    public static let simpleDiscussionRenderer = ExperimentalFeature(state: .disabled)
-    public static let graphqlSpeedGrader = ExperimentalFeature(state: .disabled)
-    public static let refreshTokens = ExperimentalFeature(state: .disabled)
-    public static let newPageDetails = ExperimentalFeature(state: .disabled)
-    public static let fileDetails = ExperimentalFeature(state: .disabled)
+    public static var allEnabled: Bool {
+        get { allFeatures.filter({ $0.isEnabled == false }).count == 0 }
+        set {
+            for feature in allFeatures {
+                feature.isEnabled = newValue
+            }
+        }
+    }
+
+    public static let allFeatures: [ExperimentalFeature] = [
+        .parent3, .conferences, .favoriteGroups, .simpleDiscussionRenderer, .graphqlSpeedGrader, .refreshTokens, .newPageDetails,
+        .fileDetails, .testing,
+    ]
+
+    // Be sure to add your feature to the `allFeatures` array as well or it will not get picked up
+    // in ExperimentalFeaturesViewController.swift
+    public static let parent3 = ExperimentalFeature(remoteConfigKey: "parent3")
+    public static let conferences = ExperimentalFeature(remoteConfigKey: "conferences")
+    public static let favoriteGroups = ExperimentalFeature(remoteConfigKey: "favorite_groups")
+    public static let simpleDiscussionRenderer = ExperimentalFeature(remoteConfigKey: "simple_discussion_renderer")
+    public static let graphqlSpeedGrader = ExperimentalFeature(remoteConfigKey: "graphql_speed_grader")
+    public static let refreshTokens = ExperimentalFeature(remoteConfigKey: "refresh_tokens")
+    public static let newPageDetails = ExperimentalFeature(remoteConfigKey: "new_page_details")
+    public static let fileDetails = ExperimentalFeature(remoteConfigKey: "file_details")
+    public static let assignmentListGraphQL = ExperimentalFeature(remoteConfigKey: "assignment_list_graphql")
+    public static let testing = ExperimentalFeature(remoteConfigKey: "testing")
```

Comment:
seems like we could save ourselves a little trouble if `ExperimentalFeature` was an enum with raw String keys. `isEnabled` would always lookup from the `UserDefaults`, & `allCases` could be generated with `CaseIterable`.